File: 9.Algorithms: Dynamic Programming/0.example.py
def memoize_add_to_80():
    cache = {}
    def inner_memo(n):
        if n in cache:
            print('short time')
            return cache[n]
        else:
            print('long time')
            cache[n] = n + 80
            return cache[n]
    return inner_memo

memo = memoize_add_to_80()
print(memo(5))
print(memo(6))
print(memo(5))

File: 9.Algorithms: Dynamic Programming/0.example.js
function addTo80(n) {
    return n + 80;
}
  
addTo80(5)
  
function memoizeAddTo80() { 
    let cache = {};
    return function(n) {
        if (n in cache) {
            console.log('short time');
            return cache[n];
        } else {
            console.log('long time');
            const answer = n + 80;
            cache[n] = answer;
            return answer;
        }
    }
}

const memoized = memoizeAddTo80();
console.log(1, memoized(6))
console.log(2, memoized(6))
